### PR TITLE
Added guard

### PR DESCRIPTION
**Pull request by jsmartt**
### Description
Adds guard, which will watch for file changes and run tests automatically. It's smart enough to figure out which specific tests to run based on the file path, which is pretty cool.

You definitely don't have to use guard, but I've found it to be very helpful and a big time saver!

### Issues Resolved
(none)

### Check List
N/A

### CODE REVIEW

**Comment by aalexmonteiro** [comment]
@jsmartt , I started the guard, but when I change a file, I do not see the guard running. Maybe I did not do something necessary.

**Comment by tmiotto** [comment]
The guard is not 'bundle exec'_ing_ the rubocop

File: Gemfile [hunk 1]
```diff
@@ -1,2 +1,13 @@
 source 'https://rubygems.org'
 gemspec
+
+begin
+  if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.2.5')
```
Comment (tmiotto) [inline Gemfile]: Guard officially supports Ruby '>= 2.2.6'

**Comment by jsmartt** [comment]
@aalexmonteiro , are you still seeing it not run when it should? If so, on which file(s)?

**Comment by aalexmonteiro** [comment]
@jsmartt , I believe I can not because I use VM with vagrant as development environment . I have seen the documentation of the guard and it seems that it is necessary to make some complementary configurations to performs the guard.

**Comment by jsmartt** [comment]
I use a Vagrant VM for dev too. Double-check your ruby version so make sure it's >= 2.2.6, because guard doesn't work with older versions of Ruby. If that's your issue, I'd recommend using RVM to try it out with a later version.

**Comment by aalexmonteiro** [comment]
I'm using the RVM too with Ruby is 2.3.1.

**Comment by ricardogpsf** [comment]
The guard run with success, but it does not execute again when some file described on Guardfile is changed.
I saw that there is a issue like this on [github of Guard](https://github.com/guard/guard/issues/848)

**Comment by tmiotto** [comment]
@jsmartt  I've tested it and it is working smooth and fine. :taco:

**Comment by tmiotto** [comment]
@jsmartt @aalexmonteiro  Do you think we're ready to merge it?

**Comment by aalexmonteiro** [comment]
👍

**Pull request merged**